HealthMonitoring.SelfHost/Content/Home/home.html

```diff
@@ -1,89 +1,155 @@
+                <p class="tagsContainer"><span ng-repeat="tag in filterTags | orderBy: 'toString()'" ng-click="removeFromTagFilter(tag)" class="endpointTag">{{tag}}</span></p>
+            </div>
+        </article>
+        <article>
+            <header>Endpoints</header>
+            <p ng-if="endpoints==null">-- connection issue --</p>
+            <p ng-if="endpoints.length==0">No endpoints registered yet</p>
+            <table ng-if="endpoints.length>0" class="endpoints">
+                <tr><th>Group</th><th>Name</th><th>Status</th><th>Response time</th><th>Last check time</th><th>Tags<th></tr>
+                <tr ng-repeat="e in endpoints | orderBy: ['Group','Name'] | filter:tagFilter">
+                    <td>{{e.Group}}</td>
+                    <td><a href="/dashboard/details?id={{e.Id}}" target="_blank">{{e.Name}}</a></td>
+                    <td class="status-{{e.Status}}">{{e.Status}}</td>
+                    <td>{{formatDuration(e.LastResponseTime)}}</td>
+                    <td>{{formatDate(e.LastCheckUtc)}}</td>
+                    <td>
+                        <p class="tagsContainer"><span ng-repeat="tag in e.Tags | orderBy: 'toString()'" ng-click="addToTagFilter(tag)" class="endpointTag">{{tag}}</span></p>
+                    </td>
+                </tr>
+            </table>
+        </article>
+        <footer>Version: {{config.Dashboard.Version}}</footer>
+    </section>
+    <script>
+        var app = angular.module('app', ['angular.filter']);
+        app.controller('ctrl', function ($scope, $http, $location) {
+            $scope.config = null;
+            $scope.alerts = [];
+            $scope.filterTags = [];
+            $scope.inputFilterTag = null;
+
+            initTagsFromUrl($scope, $location);
+
+            $scope.getConfig = function () {
+                $http.get("/api/config")
+                    .success(function (response) {
+                        $scope.config = response;
+                    })
+                    .error(function (data, status) {
+                        $scope.config = { Dashboard: { Title: "-- connection issue --" }, Version: "-- connection issue --" };
+                    });
+            }
+            $scope.getEndpoints = function () {
+
+                $http.get("/api/endpoints")
+                    .success(function (response) {
+                        $scope.endpoints = response;
+                    })
+                    .error(function (data, status) {
+                        $scope.endpoints = null;
+                    });
+            }
+
+            $scope.addTagButtonClick = function () {
+                $scope.addToTagFilter($scope.inputFilterTag);
+                $scope.inputFilterTag = null;
+            };
+
+            $scope.tagFilter = function (endpoint) {
+                for (var i = 0; i < $scope.filterTags.length; i++) {
+                    if (endpoint.Tags) {
+                        if (endpoint.Tags.length > 0) {

```

Comment:
this if is not needed, unless you implemented it due to some performance issues ?